cmd/test_test.go

```diff
@@ -272,3 +276,134 @@ test_p {
 		t.Fatalf("didn't get expected %s error when inlined schema is present; got: %v", ast.TypeErr, err)
 	}
 }
+
+func TestWatchMode(t *testing.T) {
+
+	files := map[string]string{
+		"/policy.rego":      "package foo\n p := 1",
+		"/policy_test.rego": "package foo\n test_p { p == 1 }",
+	}
+
+	test.WithTempFS(files, func(root string) {
+		var buf bytes.Buffer
+
+		testParams.output = &buf
+		testParams.watch = true
+
+		done := make(chan struct{})
+		go func() {
+			opaTest([]string{root})
+			<-done
+		}()
+
+		time.Sleep(500 * time.Millisecond)
```

Comment:
This should be ample time, but could we make this even more robust by waiting for `Watching for changes ...` to be printed to `buf` inbetween each file update? We'd need to account for the test taking too long and needing an abort, though.

Reply:
_Update_: Had to change the tests to read after the updates were done. There was a race between the test reading the output buffer and the watch mode code path adding to it. The updated tests should still exercise the new code path.